pkg/api/types.go

```diff
@@ -398,6 +402,21 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts is the endpoint name that details Glusterfs topology
+	Hosts string `json:"hosts"`
+
+	// Required: Path is the Glusterfs volume path
+	Path string `json:"path"`
+
+	// Optional: mountOptions is the mount time options such as "rw" or "ro"
+	MountOpt string `json:"mountOptions,omitempty"`
```

Comment:
I don't think we should do it this way.  I think we should have one Go struct member for each option we want to support.   Here's why:
  - individual options like "ReadOnly" can have individual `description:` tags
  - a user browsing the ui in someplace like swaggerui can see the options without diving into the glusterfs.mount man page
  - we probably don't want to allow users to set all the possible glusterfs mount options, for example `direct-io-mode `.  Having individual json options is a way of whitelisting options.
  - we may need to do validation in individual options.  I'd rather that validation code not have to parse a mount option string.
  - allows us to implement an option's effect differently (for example there might be some reason why we want to mount the glusterfs read-write, tweak it, and then bind mount it into the container readonly
  - allows kubernetes API to evolve independently of glusterfs and of varying versions of kernels and node support for glusterfs.
  - consistency with other volume types that have separate struct member for "readonly"